core/converters.py

```diff
@@ -64,3 +64,19 @@ async def convert(self, ctx: commands.Context, argument: str) -> TextChannelOrTh
             raise ChannelOrThreadNotFound(argument)
 
         return result
+
+
+class OptionalGuildConverter(commands.converter.GuildConverter):
+    """A :class:`commands.converter.GuildConverter` returning no guild (None) with "0" or "global" as guild-id input."""
+    async def convert(self, ctx: commands.Context, argument: str) -> discord.Guild | None:
+        if argument.lower() in ("0", "global"):
+            return None
+        return await super().convert(ctx, argument)
+
+
+OptionalCurrentGuild = commands.parameter(
+    default=lambda ctx: ctx.guild,
+    displayed_default="<this server>",
+    converter=OptionalGuildConverter,
+)
+"""`commands.CurrentGuild` with :class:`OptionalGuildConverter`, returning None without a current guild."""
```

Comment:
Does this docstring work? Don't you need [`typing.Annotated`](https://docs.python.org/3/library/typing.html#typing.Annotated) here?

Reply:
Docstrings after variables is supported by several tools & styles, including Sphinx, VSCode...
What docstring style do you prefer?
I think `typing.Annotated` has different syntax.

Reply:
If VSCode supports it then it's ok